R.swift/Generators/StringsGenerator.swift

```diff
@@ -0,0 +1,313 @@
+        warn("Skipping string for key \(badKey) (\(filename)), format specifiers don't match for all locales: \(locales)")
+      }
+    }
+
+    return results
+  }
+
+  private static func stringLet(values: StringValues) -> Let {
+    let escapedKey = values.key.escapedStringLiteral
+    let locales = values.values
+      .map { $0.0 }
+      .filter { !$0.isNone }
+      .map { "\"\($0.description)\"" }
+      .joinWithSeparator(", ")
+
+    return Let(
+      comments: values.comments,
+      isStatic: true,
+      name: values.key,
+      typeDefinition: .Inferred(Type.StringResource),
+      value: "StringResource(key: \"\(escapedKey)\", tableName: \"\(values.tableName)\", locales: [\(locales)])"
+    )
+  }
+
+  private static func stringFunction(values: StringValues) -> Function {
+    if values.params.isEmpty {
+      return stringFunctionNoParams(values)
+    }
+    else {
+      return stringFunctionParams(values)
+    }
+  }
+
+  private static func stringFunctionNoParams(values: StringValues) -> Function {
+
+    return Function(
+      comments: values.comments,
+      isStatic: true,
+      name: values.key,
+      generics: nil,
+      parameters: [
+        Function.Parameter(name: "_", type: Type._Void)
+      ],
+      doesThrow: false,
+      returnType: Type._String,
+      body: "return \(values.localizedString)"
+    )
+  }
+
+  private static func stringFunctionParams(values: StringValues) -> Function {
+
+    let params = values.params.enumerate().map { ix, formatSpecifier -> Function.Parameter in
+      let name = "value\(ix + 1)"
+
+      if ix == 0 {
+        return Function.Parameter(name: name, type: formatSpecifier.type)
+      }
+      else {
+        return Function.Parameter(name: "_", localName: name, type: formatSpecifier.type)
+      }
+    }
+
+    let args = params.enumerate().map { ix, _ in "value\(ix + 1)" }.joinWithSeparator(", ")
+
+    return Function(
+      comments: values.comments,
+      isStatic: true,
+      name: values.key,
+      generics: nil,
+      parameters: params,
+      doesThrow: false,
+      returnType: Type._String,
+      body: "return String(format: \(values.localizedString), locale: NSLocale.currentLocale(), \(args))"
+    )
+  }
+
+}
+
+extension Locale {
+  func withFilename(filename: String) -> String {
+    switch self {
+    case .None:
+      return "'\(filename)'"
+    case .Base:
+      return "'\(filename)' (Base)"
+    case .Language(let language):
+      return "'\(filename)' (\(language))"
+    }
+  }
+}
+
+private struct StringValues {
+  let key: String
+  let params: [FormatSpecifier]
+  let tableName: String
+  let values: [(Locale, String)]
+
+  var localizedString: String {
+    let escapedKey = key.escapedStringLiteral
+
+    if tableName == "Localizable" {
+      return "NSLocalizedString(\"\(escapedKey)\", comment: \"\")"
+    }
+    else {
+      return "NSLocalizedString(\"\(escapedKey)\", tableName: \"\(tableName)\", comment: \"\")"
```

Comment:
This would be a nice improvement. It isn't very straightforward, as it effectively involves reimplementing `NSLocalizedString`. 

Since the generated `let` values in `R.string` can already be used to do this in user code, I think we can defer this feature to later.